### DISCUSSION TITLE
Dnsmasq + Dnscrypt

### DISCUSSION THREAD

**Discussion by drtiwarir** [DNSCrypt/dnscrypt-proxy#2076] [Ideas]
I am currently using Dnsmasq and Sniproxy on a DigitalOcean droplet to watch streaming videos. I recently read about Dnscrypt, so was thinking to do a new configuration, Dnsmasq + Dnscrypt.
I have seen that it is difficult to use Dnscrypt on Apple tv and Nvidia Shield tv. 
In my current setup, i have just changed DNS (my DigitalOcean droplet IP) on my devices and it works well.
Can I use Dnsmasq and forward all requests from it to Dnscrypt, I will have to use anonymised feature in Dnscrypt to hide my IP address.
Will it work as currently working with Dnsmasq and Sniproxy ?

**Comment by bcookatpcsd** [comment]
What is dnsmasq providing that dnscrypt-proxy is not? 

Are you looking for something like this:

https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Oblivious-DoH

**Reply by drtiwarir** [reply]
Thanks for your reply.
Well you can use Dnsmasq in most of devices by simply changing the DNS
My issue is to hide my IP address, can DNScrypt forward my server's IP address instead of my home IP address ?
I have tried anonymised feature in Dnscrypt, but it does not hide my IP address.
Your help is much appreciated.

**Comment by jedisct1** [comment]
Anonymized DNSCrypt hides your IP address for DNS queries.

For other kind of network protocols, such as HTTPS, you need a VPN.
